src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -456,5 +456,335 @@ public static void OverrideJsonIgnorePropertyUsingJsonPropertyNameReversed()
+            Assert.Contains(@"""Int2"":2", serialized);
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonIgnoreConditionNever_TestData))]
+        public static void JsonIgnoreConditionNever_IgnoreNullValues_True(Type type)
+        {
+            // Property should always be (de)serialized.
+            string json = @"{""Int1"":1,""MyString"":""Random"",""Int2"":2}";
+            var options = new JsonSerializerOptions { IgnoreNullValues = true };
+
+            object obj = JsonSerializer.Deserialize(json, type, options);
+            Assert.Equal(1, (int)type.GetProperty("Int1").GetValue(obj));
+            Assert.Equal("Random", (string)type.GetProperty("MyString").GetValue(obj));
+            Assert.Equal(2, (int)type.GetProperty("Int2").GetValue(obj));
+
+            string serialized = JsonSerializer.Serialize(obj, options);
+            Assert.Contains(@"""Int1"":1", serialized);
+            Assert.Contains(@"""MyString"":""Random""", serialized);
+            Assert.Contains(@"""Int2"":2", serialized);
+
+            // Property should always be (de)serialized, even when null.
+            json = @"{""Int1"":1,""MyString"":null,""Int2"":2}";
+
+            obj = JsonSerializer.Deserialize(json, type, options);
+            Assert.Equal(1, (int)type.GetProperty("Int1").GetValue(obj));
+            Assert.Null((string)type.GetProperty("MyString").GetValue(obj));
+            Assert.Equal(2, (int)type.GetProperty("Int2").GetValue(obj));
+
+            serialized = JsonSerializer.Serialize(obj, options);
+            Assert.Contains(@"""Int1"":1", serialized);
+            Assert.Contains(@"""MyString"":null", serialized);
+            Assert.Contains(@"""Int2"":2", serialized);
+        }
+
+        private class ClassWithStructProperty_IgnoreConditionNever
+        {
+            public int Int1 { get; set; }
+            [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
+            public string MyString { get; set; }
+            public int Int2 { get; set; }
+        }
+
+        private class ClassWithStructProperty_IgnoreConditionNever_Ctor
+        {
+            public int Int1 { get; set; }
+            [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
+            public string MyString { get; }
+            public int Int2 { get; set; }
+
+            public ClassWithStructProperty_IgnoreConditionNever_Ctor(string myString)
+            {
+                MyString = myString;
+            }
+        }
+
+        private static IEnumerable<object[]> JsonIgnoreConditionNever_TestData()
+        {
+            yield return new object[] { typeof(ClassWithStructProperty_IgnoreConditionNever) };
+            yield return new object[] { typeof(ClassWithStructProperty_IgnoreConditionNever_Ctor) };
+        }
+
+        [Fact]
+        public static void JsonIgnoreCondition_CaseInsensitivityWorks()
+        {
+            string json = @"{""MyString"":""Random"",""MYSTRING"":null}";
```

Comment:
How does this feature overlap with case sensitivity functionality and\or the code? I would assume no overlap.

Reply:
It doesn't. I'll rename this test to reflect that it's a random last-one-wins functionality test.